crates/runtime/src/search/types.rs

```diff
@@ -217,7 +217,9 @@ fn transpose_and_convert(
 
     for (key, vv) in column_format {
         for (i, row_values) in vv.into_iter().enumerate() {
-            rows[i].insert(key.clone(), row_values.into());
+            if !row_values.is_empty() {
```

Comment:
This changes ensures that for matches generated by two candidate generation method on columnA and columnB, a match from only columnA has the following:

```diff
"matches": {
  "columnA": "A value",
-  "columnB": []
}
```